src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -57,13 +60,14 @@ public static int KeyGen(string className, string methodName, int numParameters)
             }
         }
 
-        public CodeGenIL(IDictionary<string, IList> input, string filePath)
+        public CodeGenIL(IDictionary<string, IList> input, string filePath, ProtoCore.MSILRuntimeCore runtimeCore)
         {
             this.input = input;
             writer = new StreamWriter(filePath);
+            this.runtimeCore = runtimeCore;
         }
 
-        public IDictionary<string, object> Emit(List<AssociativeNode> astList)
+        public IDictionary<string, object> Emit(List<AssociativeNode> astList, ProtoCore.MSILRuntimeCore runtimeCore)
```

Comment:
Why does this function need to be passed the runtime core, isn't already saved as a class member?

Reply:
will remove it